pkg/sloop/ingress/kubewatcher.go

```diff
@@ -39,32 +40,47 @@ type KubeWatcher interface {
 	Stop()
 }
 
-type kubeWatcherImpl struct {
-	informerFactory informers.SharedInformerFactory
-	stopChan        chan struct{}
-	outchan         chan typed.KubeWatchResult
-	resync          time.Duration
-	outchanlock     *sync.Mutex
-	stopped         bool
-	currentContext  string
-}
-
 type crdGroupVersionResourceKind struct {
 	group    string
 	version  string
 	resource string
 	kind     string
 }
 
+type crdInformerInfo struct {
+	crd      crdGroupVersionResourceKind
+	stopChan chan struct{}
+}
+
+type kubeWatcherImpl struct {
+	informerFactory informers.SharedInformerFactory
+	stopChan        chan struct{}
+
+	crdInformers      map[crdGroupVersionResourceKind]*crdInformerInfo
+	activeCrdInformer int64
+
+	outchan        chan typed.KubeWatchResult
+	resync         time.Duration
+	protection     *sync.Mutex
+	stopped        bool
+	refreshCrd     *time.Ticker
+	currentContext string
+}
+
 var (
+	newCrdClient = func(kubeCfg *rest.Config) (clientset.Interface, error) { return clientset.NewForConfig(kubeCfg) }
```

Comment:
can we put this somewhere common so other pkg can also use? I feel this function will be used a lot in sloop

Reply:
if you're referring to the variable - `newCrdClient` this is a hook to allow UT to replace real client with mock/fake clients - sharing across packages can be a problem
if you're referring to the implementation - this is a very lightweight wrapper around the call to `NewForConfig` - this works better as an example instead of shared code